### ISSUE TITLE
CV plugin request: simple output of host BPM as CV value?

### ISSUE THREAD

**Issue by mxmilkiib** [ryukau/LV2Plugins#19]
Some plugins do not have a host tempo sync, but they do have their own BPM parameter. In hosts like Carla, this can be exposed as an input CV port, but there is no way currently to 'get' the host BPM value to 'drive' it that way that way. Might a simple plugin that does just that be possible?

**Comment by ryukau** [comment]
I'm not sure what you mean by 'drive'. Could you explain more precisely? Maybe concrete example is helpful.

**Comment by mxmilkiib** [comment]
'Drive' just as in a mechanical synonym for 'affect' or 'change'.

As the floating point number that can be sent out a CV port isn't restricted to -1 to 1 like audio. (The MOD CV plugins use this feature, and some of the AMS plugins use that for frequency.)

So I can do this;
![image](https://user-images.githubusercontent.com/108225/85642066-41903a00-b688-11ea-8818-9c66f504d624.png)

**Comment by ryukau** [comment]
Thanks for explanation. I'll work on it.

Off topic: CV_LinearMap may be useful for the patch like the one on the image. It has multiplication range of [0, 256].

**Comment by ryukau** [comment]
Added CV_TimeInfo which outputs BPM and time signature as CV signal. Below is excerpt from documentation.

---

CV_TimeInfo provides time information as CV signal.

- Playing: Gate signal that opens when playing.
- BPM: Directly outputs the value of BPM (Beats Per Minute).
- TimeSignatureUpper: Directly outputs upper numeral of time signature.
- TimeSignatureLower: Directly outputs lower numeral of time signature.

**Caution**: Do not connect following output to audio input port:

- BPM
- TimeSignatureUpper
- TimeSignatureLower

**Comment by mxmilkiib** [comment]
Haha, good point, I used CV_LinearMap briefly, but you've made so many wonderful plugins I've only got to really exploring a few so far :)

Are there meant to be those audio ports?

![image](https://user-images.githubusercontent.com/108225/85731656-99fc2180-b6f2-11ea-97b6-ea3ee7734448.png)

There's an idea though if you were interested; a metronome. There's only two very basic audio metronome plugins currently, eg-metro and MOD version C* Click, nothing that does [poly/Euclidean](https://wiki.thingsandstuff.org/Notation#Euclidean) rhythms, and nothing that does audio, CV and MIDI in one.

**Comment by ryukau** [comment]
Thanks for the report. Fixed number of output port on CV_TimeInfo.

For basic metronome, I tried to implement bar/beat/tick trigger output for CV_TimeInfo. That's what those 3 extra ports were. But it didn't work as expected because I couldn't figure out how to map MIDI ticks to frames (sample count from the start) in DPF.

For more complex metronome, I'm not planning to make one for now. Instead I'm currently working on a piano roll sequencer.
